### DISCUSSION TITLE
Change of mode to OFF and grouping

### DISCUSSION THREAD

**Discussion by nethawk2** [nliaudat/floor-heating-controller#2] [General]
Hello,
i use the controller with home assistant.
I have 2 things i would like to ask:

I use the SONOFF NSPanel to display the User Interface which works fine.
... but: If i change the mode from HEAT to OFF (to stop heating), the cover/valve stays like it is. 
IMO it would make sense to close the valve, if the mode changes from heating to OFF.
I go around this "problem" by using a home assistant automation to close the cover,
if the cover is open and the mode is OFF.

I use 3 channels in one room. With the help of the wall panel i can set the wanted temperature
of channel 1. What would be the best way to copy the wanted temperature and mode from channel 1
to channel 2 and 3 ? Is home assistant here the way to go , or directly in ESPHome/floor-heating-controller ?

Thank you
Greetings Torsten

**Comment by nliaudat** [comment]
Hi, 
For grouping, the best way to go is [hassio cover group](https://www.home-assistant.io/integrations/group/)

To set position on OFF mode, you can change the script_termostat_check.yaml in esphome : 

```
  - id: TH1_check
    then:
         - lambda: |-
                if (id(${ch1_thermostat_name}_thermostat).mode == CLIMATE_MODE_OFF) {return;}
                ESP_LOGD("main", "TH1_check triggered");
                // check if any others actuator is running. If yes, stop action and return. 
                if (id(CH2_cover).current_operation != COVER_OPERATION_IDLE) {
                       ESP_LOGD("main", "CH2 running cause CH1 cancel operation");
                       return;
                } 
```
replace the "return" with :
```

                    auto call = id(CH1_cover).make_call();
                    call.set_position( 0 ); //target_position
                    call.perform();
```

**Reply by nethawk2** [reply]
Thanks, that works.

**Comment by nethawk2** [comment]
Hello again,

i wrote a little perl script, which runs from cron to bring the settings in sync.
Please see it as a little contribution.

Greetings Torsten

[syncech.pl.txt](https://github.com/nliaudat/floor-heating-controller/files/11196358/syncech.pl.txt)
